src/LSST/control/OutputPortBits.h

```diff
@@ -0,0 +1,115 @@
+/*
+ *  This file is part of LSST M2 support system package.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_M2CELLCPP_CONTROL_OUTPUTPORTBITS_H
+#define LSST_M2CELLCPP_CONTROL_OUTPUTPORTBITS_H
+
+// System headers
+#include <functional>
+#include <map>
+#include <memory>
+
+// Project headers
+
+namespace LSST {
+namespace m2cellcpp {
+namespace control {
+
+/// Class representation of the output port bits as seen in create_output_port_bits_masks.vi.
+/// This class contains enums for the position of all bits for the output port.
+/// Unit tests in test_SimCore.cpp.
+class OutputPortBits {
+public:
+    using Ptr = std::shared_ptr<OutputPortBits>;
+
+    /// Create a OutputPortBits with a blank _bitmap
+    OutputPortBits() = default;
+
+    /// Create a OutputPortBits with a `_bitmap` set to `bitmap`.
+    OutputPortBits(uint8_t bitmap) : _bitmap(bitmap) {}
+
+    OutputPortBits(OutputPortBits const&) = default;
+    OutputPortBits& operator=(OutputPortBits const&) = default;
+    virtual ~OutputPortBits() = default;
+
+    /// These enums should match those in create_output_port_bits_masks.vi
+    /// Changing the values of these enums will likely cause hardware problems.
+    enum {
+        ILC_MOTOR_POWER_ON = 0,    ///< “ILC Motor Power On” = 0
+        ILC_COMM_POWER_ON = 1,     ///< “ILC Comm Power On” = 1
+        CRIO_INTERLOCK_ENABLE = 2, ///< “cRIO Interlock Enable” = 2
+        RESET_MOTOR_BREAKERS = 3,  ///< “Reset Motor Power Breakers” = 3
+        RESET_COMM_BREAKERS= 4,    ///< “Reset Comm Power Breakers” = 4
+        SPARE_D05 = 5,             ///< “Spare D05” = 5
+        SPARE_D06 = 6,             ///< “Spare D06” = 6
+        SPARE_D07 = 7,             ///< “Spare D07” = 7
+    };
+
+    /// Returns the output port bit mask, which is all bits in the vi
```

Comment:
in the vi --> in the function.

Reply:
I made this function, and the other one like it, as they showed up in the LabView code largely as a reference. It's odd to make a bit mask with all the bits set. It's kind of pointless, but when it shows up in some vi logic, it's useful to have a source.